### PR TITLE
chore: add improved linting

### LINKED ISSUE #349: (linting): improve linting
### Subject

Linting

### Description

After having talked with @nikolovlazar about some code standards during reviews I think that adding a few linting rules would help to make contributing more streamlined. My proposition would be:
- warning for console.logs
- error for import order
- warning for unused variables

This is only a first step which can be improved further in the future.

### PR DESCRIPTION
**Pull request by noobinthisgame**
Closes #349 

## 📝 Description
Add better linting for typescript and best-practices to ensure that code quality stays high and to help streamline the contributing process.

## ⛳️ Current behavior (updates)

Basic next and prettier linting

## 🚀 New behavior

Now additional import, typescript and eslint linting rules are added. 

## 📝 Additional Information

To enable the the new rules I had to install a few packages:
-  "eslint-import-resolver-typescript": "^2.5.0"
-  "eslint-plugin-import": "^2.25.4"
-  "@typescript-eslint/eslint-plugin": "^5.12.1"

Also the files changes are quite a lot because of the rules hitting and fixes I had to apply to resolve those.

### CODE REVIEW

File: src/components/chakra-next-image.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import { chakra } from '@chakra-ui/system'
 import { Flex } from '@chakra-ui/layout'
 import type { FlexProps } from '@chakra-ui/layout'
-import NextImage, { ImageProps, ImageLoaderProps } from 'next/image'
+import { chakra } from '@chakra-ui/system'
+import NextImage, { ImageLoaderProps, ImageProps } from 'next/image'
```
Comment (nikolovlazar) [inline src/components/chakra-next-image.tsx:4]: Let's add `type ImageLoaderProps, type ImageProps` to the props imports. We can also do that to the `FlexProps` above and merge the first two lines together.

**Review by nikolovlazar — APPROVED** [review]
LGTM! Aside from that little thing I commented on 😁

File: src/components/chakra-next-image.tsx (hunk 1 above)
Comment (noobinthisgame) [reply src/components/chakra-next-image.tsx:4]: Right good catch will do later :)

Comment (noobinthisgame) [reply src/components/chakra-next-image.tsx]: I tried out the inline syntax, but it wouldn't let the application build. Maybe we need a newer next version.

**Comment by noobinthisgame** [comment]
@nikolovlazar I just thought about this and would say that we try to stall this until the structure PR is merged because this might be conflicting quite a bit with the other one and has less priority.

**Comment by nikolovlazar** [comment]
Yeah @noobinthisgame. If you think that it's conflicting with the new restructure PR, feel free to mark it as draft, so we can revisit it later on.

**Pull request merged**